modules/AWS/rds/main.tf

```diff
@@ -40,20 +40,29 @@ module "db" {
   allocated_storage = var.allocated_storage
   iops              = var.iops
 
-  name                   = var.product
-  username               = local.db_master_usr
-  create_random_password = true
-  random_password_length = 12
-  port                   = 5432
+  name     = var.product
+  username = local.db_master_username
+  password = local.db_master_password
+  port     = 5432
 
   subnet_ids             = var.vpc.private_subnets
   vpc_security_group_ids = [module.security_group.security_group_id]
 
   maintenance_window = "Mon:00:00-Mon:03:00"
   backup_window      = "03:00-06:00"
 
+  # Snapshot settings
+  snapshot_identifier         = var.snapshot_identifier
+  allow_major_version_upgrade = true
```

Comment:
Is this necessary for snapshot restore?

Reply:
This is needed to avoid put additional restriction to match the major version of database with what is in snapshot. 
The process of restore the data from snapshot is first terraform provisions the database instance, and then restores snapshot. If the major version is not matched, then terraform tries to match them with upgrade/downgrade the major engine version. IF this is not possible, then an error will be raised follow by install interruption. 

To avoid it, we can set this flag true and let RDS match the versions. However, I feel we can do it only when snapshot is provided to make the impact more limited. Something like:
```
allow_major_version_upgrade = var.snapshot_identifier != null
```